src/tike/operators/cupy/fresnelspectprop.py

```diff
@@ -0,0 +1,131 @@
+"""Defines a free-space propagation operator based on the CuPy FFT module."""
+
+__author__ = "Ashish Tripathi"
+__copyright__ = "Copyright (c) 2024, UChicago Argonne, LLC."
+
+import typing
+
+import numpy.typing as npt
+import numpy as np
+
+from .cache import CachedFFT
+from .operator import Operator
+
+
+class FresnelSpectProp(CachedFFT, Operator):
+    """Fresnel spectrum propagation (short range) using CuPy.
+
+    Take an (..., W, H) compelx array representing a wavefront and propagate.
+
+    Attributes
+    ----------
+    pixel_size : float
+        The realspace size of a pixel in meters
+    distance : float
+        The realspace propagation distance in meters
+    wavelength : float
+        The wavelength of the light in meters
+
+    Parameters
+    ----------
+    nearplane: (..., W, H) complex64
+        The wavefronts before propagation.
+    farplane: (..., W, H) complex64
+        The wavefronts after propagation.
+    """
+
+    def __init__(
+        self,
+        norm: str = "ortho",
+        pixel_size: float = 1.0,
+        distance: float = 1.0,
+        wavelength: float = 1.0,
```

Comment:
Maybe there should be no default values.

Reply:
It might be useful if we at least use "order of magnitude" defaults? e.g. distance = 1e-6 meters, wavelength = 1e-9 meters, pixel size = 100e-6 meters?